openstef/model/regressors/arima.py

```diff
@@ -0,0 +1,163 @@
+# SPDX-FileCopyrightText: 2017-2022 Alliander N.V. <korte.termijn.prognoses@alliander.com> # noqa E501>
+#
+# SPDX-License-Identifier: MPL-2.0
+"""This module contains the SARIMAX regressor wrapper around statsmodels implementation."""
+import numpy as np
+import pandas as pd
+import statsmodels.api as sm
+
+from sklearn.metrics import r2_score
+from sklearn.model_selection import TimeSeriesSplit
+from openstef.model.regressors.regressor import OpenstfRegressor
+
+
+class ARIMAOpenstfRegressor(OpenstfRegressor):
+    """Wrapper around statmodels implementation of (S)ARIMA(X) model.
+
+    To make prediction, it needs to update its historic, applying the fitted parameters to new data unrelated to the
+    original training data.
+
+    """
+
+    def __init__(
+        self,
+        backtest_max_horizon=1440,
+        order=(0, 0, 0),
+        seasonal_order=(0, 0, 0, 0),
+        trend=None,
+    ):
+        self.backtest_max_horizon = backtest_max_horizon
+        self.order = order
+        self.seasonal_order = seasonal_order
+        self.trend = trend
+
+    def fit(self, x, y, **kwargs):
+        exog = x
+        dates = x.index
+        self.model_ = sm.tsa.arima.ARIMA(
+            y,
+            exog,
+            dates=dates,
+            order=self.order,
+            seasonal_order=self.seasonal_order,
+            trend=self.trend,
+        )
+        self.results_ = self.model_.fit()
+        self.feature_in_names_ = list(x.columns)
+        return self
+
+    def update_historic(self, y_past, x_past):
```

Comment:
Also here I do not understand the historic terminology, do you mean history or maybe historic data?